resolver-dns/src/main/java/io/netty/resolver/dns/DnsResolveContext.java

```diff
@@ -1052,6 +1052,11 @@ private void finishResolve(Promise<List<T>> promise, Throwable cause) {
                         ReferenceCountUtil.safeRelease(item);
                     }
                 }
+            } else {
```

Comment:
since users can pass in the `Promise` and cancel/complete it themselves this is racy. The cancellation maybe more likely to occur but in that case should we just always go through the above path?

Reply:
Yeah maybe this is even better...